content/en/blog/_posts/Introducing-Kueue.md

```diff
@@ -0,0 +1,199 @@
+
+For more details, take a look at the [API concepts documentation](https://sigs.k8s.io/kueue/docs/concepts). While the three APIs may look overwhelming, 
+most of Kueue’s operations are centered around ClusterQueue; the ResourceFlavor and LocalQueue APIs are mainly organizational wrappers.
+
+### Example Use Case
+Imagine the following setup for running batch workloads on a Kubernetes cluster on the cloud: 
+- We have two types of autoscaled node groups that differ on their provisioning policies: spot and on-demand. The nodes of each group are 
+  differentiated by the label `instance-type = {spot | ondemand}`. Moreover, since not all jobs can tolerate running on spot nodes, the nodes 
+  are tainted with spot=true:NoSchedule.
+- To strike a balance between cost and resource availability, we want jobs to use up to 1000 cores of on-demand nodes, then use up to 2000 cores 
+  of spot nodes.
+
+As batch admins, we define two ResourceFlavors that represent the two types of nodes
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ResourceFlavor
+metadata:
+  name: ondemand
+  labels:
+    instance-type: ondemand 
+---
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ResourceFlavor
+metadata:
+  name: spot
+  labels:
+    instance-type: spot
+taints:
+- effect: NoSchedule
+  key: spot
+  value: "true"
+```
+
+Then we define the quotas by creating a ClusterQueue as follows
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ClusterQueue
+metadata:
+  name: research-pool
+spec:
+  namespaceSelector: {}
+  resources:
+  - name: "cpu"
+    flavors:
+    - name: ondemand
+      quota:
+        min: 1000
+    - name: spot
+      quota:
+        min: 2000
+```
+
+Note that the order of flavors in the ClusterQueue resources matters: Kueue will attempt to fit jobs in the available quotas according to 
+the order unless the job has an explicit affinity to specific flavors. 
+
+For each namespace, we define a LocalQueue that points to the ClusterQueue above.
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: LocalQueue
+metadata:
+  name: training
+  namespace: team-ml
+spec:
+  clusterQueue: research-pool
+```
+
+Admins create the above setup once. As batch users, we are able to find the queues we are allowed to submit to by listing the `LocalQueues` 
+in our namespace. The command is similar to the following: `kubectl get -n my-namespace localqueues`
+
+To submit a job, we simply set the `kueue.x-k8s.io/queue-name` annotation as follows:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  annotations:
+    kueue.x-k8s.io/queue-name: main
+spec:
+  parallelism: 3
+  completions: 3
+  template:
+    spec:
+      tolerations:
+      - key: spot
+        operator: "Exists"
+        effect: "NoSchedule"
+      containers:
+      - name: dummy-job
+        image: gcr.io/k8s-staging-perf-tests/sleep:latest
+        args: ["30s"]
+        resources:
+          requests:
+            cpu: 1
```

Comment:
What's our view on setting a resource limit for CPU? I'm mindful that some workloads will impact other things running on the node if not otherwise inhibited.

Reply:
Setting limits is a good practice in general, especially for memory, but it is not quite relevant here, so I left it out.